### PR TITLE
PR: Restore CircleCI

### LINKED ISSUE #54: Restore continuous integration
@steff456, please restore continuous integration to this repo with CircleCI (as it was before). For that you need to add `.circleci/config.yml` to the root of the repo.

Please take as a template these files:

https://github.com/spyder-ide/spyder-kernels/blob/master/.circleci/config.yml
https://github.com/spyder-ide/spyder-kernels/blob/master/ci/install.sh
https://github.com/spyder-ide/spyder-kernels/blob/master/ci/runtests.sh

run our tests only for Python 2.7 and 3.7, and update

https://github.com/spyder-ide/spyder-plugin-cookiecutter/blob/master/requirements.txt

by removing old dependencies (like versioneer and ciocheck) and removing the unnecessary pins there.

Thanks!

### PR DESCRIPTION
**Pull request by steff456**
Fixes: #54

### CODE REVIEW

File: ci/install.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
Comment (ccordoba12) [inline ci/install.sh]: Please make this script executable and change this shebang to be `#!/bin/bash -ex`

File: ci/install.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Install Miniconda
+if [ $(uname) == Linux ]; then
+    wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+else
+    wget https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh -O miniconda.sh
+fi
```
Comment (ccordoba12) [inline ci/install.sh]: Let's remove this `if` and only install Miniconda3 for Linux (Circle doesn't support macOS on its free version),

File: ci/install.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+# Make new conda environment with required Python version
+conda create -y -n test python=$PYTHON_VERSION
+conda activate test
+
+# Install dependencies
+conda install -y -q --file requirements/requirements.txt
+
+# Install test dependencies
+conda install -y -q nomkl
```
Comment (ccordoba12) [inline ci/install.sh]: This is not necessary, so please remove it.

File: ci/install.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+fi
+bash miniconda.sh -b -p $HOME/miniconda
+source $HOME/miniconda/etc/profile.d/conda.sh
+
+# Make new conda environment with required Python version
+conda create -y -n test python=$PYTHON_VERSION
+conda activate test
+
+# Install dependencies
+conda install -y -q --file requirements/requirements.txt
```
Comment (ccordoba12) [inline ci/install.sh]: Let's change this to

    conda install -c spyder-ide -y -q requirements/conda.txt
    conda install -c spyder-ide -y -q requirements/tests.txt

File: ci/runtests.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (ccordoba12) [inline ci/runtests.sh]: Please also make this an executable script, and change this shebang to be `#!/bin/bash -ex`

File: ci/runtests.sh [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+source $HOME/miniconda/etc/profile.d/conda.sh
+conda activate test
+
+pip install -e .
+
+pytest -x -vv .
+
+if [ $? -ne 0 ]; then
+    exit 1
+fi
```
Comment (ccordoba12) [inline ci/runtests.sh]: This `if` is not necessary with the addition of `-ex` to the shebang, so please remove it (`-ex` makes a bash script to fail if any of the commands inside it fails).

File: requirements/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,8 @@
+cookiecutter
```
Comment (ccordoba12) [inline requirements/requirements.txt]: Let's rename this file to `conda.txt` and only leave `cookiecutter` and `entrypoints` on it.

Also, please create a new file called `tests.txt` and add there `pytest` and `pytest-cookies`.

**Review by ccordoba12 — COMMENTED** [review]
@steff456, tests should be working now (I re-added the Circle public key to be sure).

I also left you several comments so you can finish this one.

File: ci/install.sh [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+bash miniconda.sh -b -p $HOME/miniconda
+source $HOME/miniconda/etc/profile.d/conda.sh
+
+# Make new conda environment with required Python version
+conda create -y -n test python=$PYTHON_VERSION
+conda activate test
+
+# Install dependencies
+conda install -c anaconda -y -q requirements/conda.txt
+conda install -y -q requirements/tests.txt
```
Comment (ccordoba12) [inline ci/install.sh]: Sorry, this needs to be

    conda install -c spyder-ide -y -q --file requirements/conda.txt
    conda install -c spyder-ide -y -q --file requirements/tests.txt

Comment (steff456) [reply ci/install.sh]: The packages `entrypoints` and `cookiecutter` are not in spyder-ide channel

Comment (ccordoba12) [reply ci/install.sh]: Yes, they are:

https://anaconda.org/spyder-ide/repo

The thing that was missing from the above commands was the `--file` option, to install the packages in conda.txt and tests.txt.

**Comment by steff456** [comment]
@ccordoba12 I'm not sure if these dependencies should be available in the spyder channel as well

image

**Comment by ccordoba12** [comment]
Sorry for the delay on addressing this one. I'm fixing it right now.

File: ci/runtests.sh [hunk 9]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -ex
+
+source $HOME/miniconda/etc/profile.d/conda.sh
+conda activate test
+
+pip install -e .
```
Comment (ccordoba12) [inline ci/runtests.sh]: This is what's making our tests fail now and it's not necessary. Please remove it.

**Comment by ccordoba12** [comment]
Thanks @steff456! Things are finally working again!

Please squash all your commits in a single one, so I can merge this one.

**Comment by ccordoba12** [comment]
You should have squashed all commits before merging (see my message above). Please don't forget to do that for next time.

**Comment by steff456** [comment]
> You should have squashed all commits before merging (see my message above). Please don't forget to do that for next time.

I used the new option from github `squash and merge` if you see master branch only has one commit from this PR

**Comment by ccordoba12** [comment]
Ok, didn't see that. Thanks!

**Pull request merged**